files/default/neutron-ha-tool.py

```diff
@@ -654,10 +654,14 @@ def migrate_router(qclient, router_id, agent, target,
     qclient.remove_router_from_l3_agent(agent['id'], router_id)
     LOG.debug("Removed router from agent=%s" % agent['id'])
 
-    # ensure it is removed or log an error
+    # ensure it is removed
     if router_id in list_routers_on_l3_agent(qclient, agent['id']):
-        raise RuntimeError("Failed to remove router_id=%s from agent_id=%s" %
-                           (router_id, agent['id']))
+        qclient.remove_router_from_l3_agent(agent['id'], router_id)
+        LOG.debug("Removed (again) router from agent=%s" % agent['id'])
```

Comment:
Not sure that I like the workaround : ). Actually the resource agent is retrying that, and when I did some manual test this worked. So that means that if I am not wrong, the first delete raised and exception somewhere.

But if it is not raising anything, I have two comments:

* Add a reference of the bsc# bug report in the comment in the code and in the commit. Be clear in the code that this is a workaround that needs to be reverted once the real bug is fixed (or SOC6 dies)

* I wonder if "Removed (again)" is a bit misleading here. Maybe a WARNING, instead of a DEBUG and with a different text, like "The router was not correctly deleted, retrying" or something